### ISSUE TITLE
Corporate proxy checks pages

### ISSUE THREAD

**Issue by alexfriman** [erebe/wstunnel#8]
They check if a real page is there on 443 port, before allowing the connection. I can see in output "Failed reading: invalid header type" on the server side, and "ERROR :: Connection not allowed by the proxy" on the client side. 
Is it possible to give them a sample page instead of generating an error?

Cheers, Alex

**Comment by erebe** [comment]
Can you start both the client and the server in verbose mode and give the output please ?
(-v in order to get the verbose mode)

**Comment by alexfriman** [comment]
Sure!

root@server:~# wstunnel --server wss://0.0.0.0:443 -r 127.0.0.1:4433 -v
Starting server with opts WsServerInfo {useTls = True, host = "0.0.0.0", port = 443}
WAIT for TLS connection on 0.0.0.0:443
wstunnel: HandshakeFailed (Error_Misc "sign failed: RSAError SignatureTooLong")
wstunnel: HandshakeFailed (Error_Misc "sign failed: RSAError SignatureTooLong")

---
[root@client]# wstunnel -L 9999:127.0.0.1:4433 -p proxyhost:3129 wss://server:443 -v
WAIT for tcp connection on 127.0.0.1:9999
DEBUG :: Oppening tcp connection to proxy ProxySettings {host = "proxyhost", port = 3129, credentials = Nothing}
DEBUG :: Doing tls Handshake
DEBUG :: Closing tcp connection to proxy ProxySettings {host = "proxyhost", port = 3129, credentials = Nothing}
ERROR :: Cannot do tls handshake with the server
DEBUG :: ====
HandshakeFailed (Error_Misc "user error (unexpected type received. expecting handshake and got: Alert [(AlertLevel_Fatal,InternalError)])")

**Comment by erebe** [comment]
Strange error, do you get always the same ?

While trying to connect with the client can you do on the server a
`tcpdump -i any port 443 -w server.pcap`
and on the client a 
`tcpdump -i any port 3129 -w client.pcap`
and send me those 2 pcap files ?

You can send me those pcap files by mail if you don't want to post it on github, they should not be too big

**Comment by erebe** [comment]
Hello,
Thanks I got your pcaps files.

Quick answer: Use the binary provided by the released https://github.com/erebe/wstunnel/releases/tag/1.2

So the issue is due because wstunnel (server side) is picking a cipher suite which is not valid with the certificate key length.
By default the cipher suite `TLS_RSA_WITH_AES_128_CBC_SHA` should be used, but in your case the server is selecting `TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256`.

The only way I managed to reproduce this behavior is by re-building wstunnel with an other stack LTS (or directly with cabal), so I am assuming that you have built wstunnel yourself ? Am I right ?

Anyway this is a regression (caused by https://github.com/snoyberg/conduit/commit/36a2d61c24956601fe48f52e7527d4678fe4e2b0) which I should fix for future release.

But in the mean time, if you want to use wstunnel with tls, use the binary provided in the release https://github.com/erebe/wstunnel/releases/tag/1.2, it has been build without the patch link above and will use a cipher that is valid with the certificate key length.

Let me know if you indeed or not use a own built wstunnel.
